shuup/core/api/basket.py

```diff
@@ -547,6 +550,21 @@ def _handle_setting_address(self, request, attr_field):
             data = BasketSerializer(request.basket, context=self.get_serializer_context()).data
             return Response(data, status=status.HTTP_200_OK)
 
+    @schema_serializer_class(ExtraDataSerializer)
+    @detail_route(methods=['post'])
+    def set_extra_data(self, request, *args, **kwargs):
+        self.process_request()
+        try:
+            extra_data = {}
+            for k, v in request.data.items():
+                extra_data[k] = v
+            request.basket.extra_data = extra_data
+            request.basket.save()
+            data = BasketSerializer(request.basket, context=self.get_serializer_context()).data
+            return Response(data, status=status.HTTP_200_OK)
+        except:
+            return Response({"error", "Bad Request"}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Why do you need this separate set_extra_data endpoint? Wouldn't it make more sense to have a single `update` endpoint, which can update the extra_data, billing_address, shipping_address, codes, etc. whichever is set in the PATCH request? E.g. if the client calls `PATCH /api/shuup/basket/BASKETNAME` with argument like `{"extra_data": {"the": "extra data", "to": "set"}}`, it would set just the extra_data and similarly you could set only the billing address by providing just that in the input. That would also allow updating all aspects of the basket in a single request.